mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,302 @@
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import Row from "../../spacing/Row";
+import Col from "../../spacing/Col";
+import Select from "../../elements/Select";
+import DropdownItem from "../../elements/DropdownItem";
+import { getDisplayColumnHeader } from "../../../utils/columnHeaders";
+import MultiSelectButtons from "../../elements/MultiSelectButtons";
+import useSendEditOnClick from "../../../hooks/useSendEditOnClick";
+import TextButton from "../../elements/TextButton";
+import Input from "../../elements/Input";
+import { useDebouncedEffect } from "../../../hooks/useDebouncedEffect";
+import '../../../../css/taskpanes/SplitTextToColumns/SplitTextToColumns.css'
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+
+interface SplitTextToColumnsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    gridState: GridState;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    dfNames: string[];
+    startingColumnID?: ColumnID
+}
+
+type DelimiterObj = {
+    included: boolean,
+    delimiter: string
+}
+
+const defaultDelimitersObj: Record<string, DelimiterObj> = {
+    'Comma': {
+        included: false,
+        delimiter: ','
+    },
+    'Dash': {
+        included: false,
+        delimiter: '-'
+    },
+    'Tab': {
+        included: false,
+        delimiter: '\t'
+    },
+    'Space': {
+        included: false,
+        delimiter: ' '
+    },
+    'Other': {
+        included: false,
+        delimiter: ''
+    }
+}
+
+/* 
+    This taskpane allows users to split a column into multiple columns 
+    by separating on a delimeter
+*/
+const SplitTextToColumnsTaskpane = (props: SplitTextToColumnsTaskpaneProps): JSX.Element => {
+
+    // Use the selected column, unless its the index column, then just use the first column
+    const startingColumnIndex = props.gridState.selections[0].startingColumnIndex >= 0 ? props.gridState.selections[0].startingColumnIndex : 0
+
+    // Make sure that there is data in the sheet before trying to access the column_id, so we don't crash the sheet
+    const isSheetEmpty = props.sheetDataArray[props.gridState.sheetIndex] === undefined || props.sheetDataArray[props.gridState.sheetIndex].data[startingColumnIndex] === undefined
+ 
+    // Get the starting column id
+    let startingColumnID = undefined
+    if (props.startingColumnID !== undefined) {
+        startingColumnID = props.startingColumnID
+    } else if (!isSheetEmpty) {
+        startingColumnID = props.sheetDataArray[props.gridState.sheetIndex].data[startingColumnIndex].columnID
+    } 
+
+    const {params, setParams, loading, edit, editApplied} = useSendEditOnClick<SplitTextToColumnsParams, undefined>(
```

Comment:
Hook should be defined at the stop of the file. If you have to do work to get data to call this, do it in a function for clarity